R/app_server.R

```diff
@@ -40,13 +40,22 @@ app_server <- function(input, output, session) {
   # reactive variables
 
   rawdata <- reactive({
-    req(input$upload)
+    req(input$upload)  # Ensure the file is uploaded
```

Comment:
Thank you so much for checking out searchbuildR! I am wondering why this error occurs for you. Should this not be handled by line 43 `req()` without any further checks necessary?

Reply:
That would make sense to me! I am not sure but it seems to be because of the switch() which is looking for either txt or ris file. Shiny can be a bit weird about how it runs code (not always in what we think is "order")